### PR TITLE
*: Refine MutableSupport and DeltaMergeDefines

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #9673 

Problem Summary:

After https://github.com/pingcap/tiflash/pull/9608, the storage format of strings depends on `STORAGE_FORMAT_CURRENT`.

But `MutableSupport::tidb_pk_column_string_type` is a static variable that is only initialized once at program startup. Its value depends on the initialization value of `STORAGE_FORMAT_CURRENT` in the code, and modifying the configuration will not change it.

And in order to test different storage formats, some unit-tests will dynamically modify the configuration of storage formats.

In summary, we need to dynamically obtain the object of `DataTypeString` based on the value of `STORAGE_FORMAT_CURRENT`.

 
### What is changed and how it works?

- Dynamically obtain the object of `DataTypeString` based on the value of `STORAGE_FORMAT_CURRENT`in
  - `MutableSupport.cpp` - `MutSup::getTiDBPkColumnStringType()`
  - `DeltaMergeDefines.cpp` - `getExtraStringHandleColumnDefine()`

- Also this PR refine code in `MutableSupport.*` and `DeltaMergeDefines.*`.
  - Remove some useless.
  - Remove some deplicated.
  - Remove some macros.

```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/MutableSupport.h [hunk 1]
```diff
@@ -14,84 +14,53 @@
...
-        static const UInt8 DEL_MASK = (INTERNAL_DEL | DEFINITE_DEL);
-        static const UInt8 DATA_MASK = ~DEL_MASK;
-
-        static UInt8 getData(UInt8 raw_data) { return raw_data & DATA_MASK; }
-
-        static bool isDel(UInt8 raw_data) { return raw_data & DEL_MASK; }
+    inline static const String delta_tree_storage_name = "DeltaMerge";
 
-        static bool isDefiniteDel(UInt8 raw_data) { return raw_data & DEFINITE_DEL; }
+    inline static constexpr ColumnID tidb_pk_col_id = -1;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/MutableSupport.h:44]:
```suggestion
    inline static constexpr ColumnID extra_handle_id = -1;
```
"pk_col_id" is not accurate. Align with the name "extra_handle_id" with that in tidb-server?
https://github.com/pingcap/tidb/blob/d13e52ed6e22cc5789bed7c64c861578cd2ed55b/pkg/meta/model/table.go#L31-L38

File: dbms/src/Storages/MutableSupport.h [hunk 2]
```diff
@@ -14,84 +14,53 @@
...
+    inline static constexpr ColumnID extra_table_id_col_id = -3;
+    inline static constexpr ColumnID version_col_id = -1024;
+    inline static constexpr ColumnID delmark_col_id = -1025;
+    inline static constexpr ColumnID invalid_col_id = -10000;
 
-        static UInt8 genDelMark(bool internal_del, bool definite_del, UInt8 src_data)
-        {
-            return (internal_del ? INTERNAL_DEL : NONE) | (definite_del ? DEFINITE_DEL : NONE) | getData(src_data);
-        }
+    inline static const String tidb_pk_column_name = "_tidb_rowid";
```
Comment (JaySon-Huang) [inline dbms/src/Storages/MutableSupport.h:50]:
```suggestion
    inline static const String extra_handle_column_name = "_tidb_rowid";
```

File: dbms/src/Storages/MutableSupport.h [hunk 3]
```diff
@@ -14,84 +14,53 @@
...
+    inline static const String delmark_column_name = "_INTERNAL_DELMARK";
+    inline static const String extra_table_id_column_name = "_tidb_tid";
 
-        static UInt8 genDelMark(bool internal_del, bool definite_del = false)
-        {
-            return genDelMark(internal_del, definite_del, 0);
-        }
-    };
+    ALWAYS_INLINE static const DataTypePtr & getTiDBPkColumnIntType();
+    ALWAYS_INLINE static const DataTypePtr & getTiDBPkColumnStringType();
```
Comment (JaySon-Huang) [inline dbms/src/Storages/MutableSupport.h:56]:
```suggestion
    ALWAYS_INLINE static const DataTypePtr & getExtraHandleColumnIntType();
    ALWAYS_INLINE static const DataTypePtr & getExtraHandleColumnStringType();
```

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**